tests/e2e/tests/pilot/performance/serviceentry_test.go

```diff
@@ -0,0 +1,346 @@
+				Value:   marshaledServiceEntry(servicePorts[i], backendPorts[i], host),
+			},
+		})
+
+	}
+	t.cachedResponse = envelopes
+	t.cachedResponseMux.Unlock()
+	return envelopes
+}
+
+func marshaledServiceEntry(servicePort, backendPort int, host string) []byte {
+	serviceEntry := &networking.ServiceEntry{
+		Hosts: []string{host},
+		Ports: []*networking.Port{
+			{
+				Name:     "http",
+				Number:   uint32(servicePort),
+				Protocol: "http",
+			},
+		},
+		Location:   networking.ServiceEntry_MESH_INTERNAL,
+		Resolution: networking.ServiceEntry_STATIC,
+		Endpoints: []*networking.ServiceEntry_Endpoint{
+			{
+				Address: "127.0.0.1",
+				Ports: map[string]uint32{
+					"http": uint32(backendPort),
+				},
+				Labels: map[string]string{"label": randName()},
+			},
+		},
+	}
+
+	se, err := proto.Marshal(serviceEntry)
+	if err != nil {
+		log.Fatalf("marshaling service entry %s\n", err)
+	}
+	return se
+}
+
+func generateGateway(req *mcp.MeshConfigRequest) *mcpserver.WatchResponse {
+	return &mcpserver.WatchResponse{
+		Version: req.GetVersionInfo(),
+		TypeURL: req.GetTypeUrl(),
+		Envelopes: []*mcp.Envelope{
+			{
+				Metadata: &mcp.Metadata{
+					Name:       fmt.Sprintf("gateway-%s", randName()),
+					CreateTime: createTime(),
+				},
+				Resource: &types.Any{
+					TypeUrl: req.GetTypeUrl(),
+					Value:   marshaledGateway(),
+				},
+			},
+		},
+	}
+}
+
+func marshaledGateway() []byte {
+	gateway := &networking.Gateway{
+		Servers: []*networking.Server{
+			&networking.Server{
+				Port: &networking.Port{
+					Name:     "http-8099",
+					Number:   8099,
+					Protocol: "http",
+				},
+				Hosts: []string{"*"},
+			},
+		},
+	}
+
+	gw, err := proto.Marshal(gateway)
+	if err != nil {
+		log.Fatalf("marshaling gateway %s\n", err)
+	}
+	return gw
+
+}
+
+func createTime() *types.Timestamp {
+	timeStamp, err := types.TimestampProto(time.Now())
+	if err != nil {
+		log.Fatalf("creating proto timestamp %s\n", err)
+	}
+	return timeStamp
+}
+func randName() string {
+	name := uuid.New()
+	return name.String()
+}
+
+func availablePorts(n int) (ports []int) {
+	for i := 0; i < n; i++ {
+		addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+		if err != nil {
+			log.Fatalf("generating ports %s\n", err)
+		}
+
+		l, err := net.ListenTCP("tcp", addr)
+		if err != nil {
+			log.Fatalf("generating ports %s\n", err)
+		}
+		defer l.Close()
+		ports = append(ports, l.Addr().(*net.TCPAddr).Port)
+	}
+	return ports
+}
+
+func initLocalPilotTestEnv(t *testing.T, mcpPort, grpcPort, debugPort int) util.TearDownFunc {
+	mixerEnv.NewTestSetup(mixerEnv.PilotMCPTest, t)
+	debugAddr := fmt.Sprintf("127.0.0.1:%d", debugPort)
+	grpcAddr := fmt.Sprintf("127.0.0.1:%d", grpcPort)
+	_, tearDown := util.EnsureTestServer(addMcpAddrs(mcpPort), setupPilotDiscoveryHTTPAddr(debugAddr), setupPilotDiscoveryGrpcAddr(grpcAddr))
+	return tearDown
+}
+
+func runMcpServer(param *testParam) (*mockmcp.Server, error) {
```

Comment:
I didn't realize that server existed, looking at it now it does seem like a lot of duplication. I will update this to use the server you mentioned above to avoid duplication. Also I might have to go over some of the other MCP tests to make sure that they are using the server you mentioned so that I can get rid of the duplicate server altogether.